fasm/parse_fasm.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020  The SymbiFlow Authors.
+#
+# Use of this source code is governed by a ISC-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier: ISC
+
+from ctypes import CDLL, POINTER, CFUNCTYPE, c_char
+import os
+from sys import byteorder
+from fasm.model import \
+    SetFasmFeature, Annotation, FasmLine, tag_to_value_format
+import fasm.tags as tags
+import platform
+from pathlib import Path
+
+if platform.system() == 'Darwin':
+    parse_fasm_lib = "libparse_fasm.dylib"
+else:
+    parse_fasm_lib = "libparse_fasm.so"
+
+here = Path(os.path.dirname(os.path.realpath(__file__)))
+parse_fasm = CDLL(str(here / parse_fasm_lib))
+parse_fasm.from_string.restype = POINTER(c_char)
+parse_fasm.from_file.restype = POINTER(c_char)
+
+# The following functions decode the binary format generated by parse_fasm.
+# This is a lightweight binary format designed to be fast to decode.
+
+
+def get_header(tags, data, i):
+    """ Match a tag and retrieve length from the header. """
```

Comment:
Please mention that None is return and index is not advanced if tag doesn't match.

Reply:
Done.